src/util/nidmap.c

```diff
@@ -1139,8 +1144,8 @@ int prrte_util_decode_ppn(prrte_job_t *jdata,
     prrte_buffer_t bucket;
 
     /* reset any flags */
-    for (m=0; m < prrte_node_pool->size; m++) {
-        if (NULL != (node = (prrte_node_t*)prrte_pointer_array_get_item(prrte_node_pool, m))) {
+    for (m=0; m < jdata->map->num_nodes; m++) {
```

Comment:
You have to cycle until jdata->map->nodes->size - although the pointer array _should_ be left-packed, it isn't technically guaranteed

Reply:
Done.